### ISSUE TITLE
cosine similarity

### ISSUE THREAD

**Issue by datistiquo** [UKPLab/sentence-transformers#549]
I am a little bit confused about the handling with cosine similarity/distance like in the  CosineSimilarityLoss.  When training the model with MSE from the cosine of u and v, this value can range between -1 and 1. So very similiar u,v have value 1 and dissimiliar have -1.  Then in some cases when handling cosine distance I often see that you have a "Minus" in front of the cosine like with tf.keras: https://www.tensorflow.org/api_docs/python/tf/keras/losses/cosine_similarity

Maybe I do not understand why one uses a minus, but what is the difference between you handling (MSE with plain cosine without any extra minus) and the other handlings with a minus?

ALso, how and why is  the MSE Loss used with cosine similarity? Because when cosine equals -1 how could you minimize then theloss for non-duplicates? So if label is 0 and cosine -1 how could you minimze MSE loss. Better would be if you have just a cosine between 0 and 1, right?
